src/listeners/interactionCreate.ts

```diff
@@ -75,26 +75,41 @@ export default class InteractionCreateListener extends Listener
 		if (!account) return this.respondAccountSignup(id, token);
 
 		const coinsOption = options.find(option => option.name === "coins");
-		if (!coinsOption || typeof coinsOption.value !== "number" || coinsOption.value <= 0)
+		if (!coinsOption || typeof coinsOption.value !== "number" || coinsOption.value < 50)
 		{
-			return this.respondEphemeral(id, token, { content: "You need to gamble at least 1 coin." });
+			return this.respondEphemeral(id, token, { content: "You need to gamble at least 50 coins." });
 		}
+		const coins = coinsOption.value;
 
-		let coins = coinsOption.value;
-
-		// Win -> double your stake, lose -> lose your stake
-		const multiplier = Math.random() < 0.05 ** 3 ? 1 : -1;
-		coins *= multiplier;
-		await Tibia.grantCoins(userId, coins);
-
-		if (multiplier < 0)
+		const multiplierOption = options.find(option => option.name === "multiplier");
+		if (!multiplierOption || typeof multiplierOption.value !== "number" || multiplierOption.value < 2)
 		{
-			return this.respondNormal(id, token, { content: `<@${userId}>, you lost your stake.` });
+			return this.respondEphemeral(id, token, { content: "You need to have a minimum multiplier of 2." });
 		}
-		else
+		const multiplier = multiplierOption ? multiplierOption.value : 2;
+
+		await Tibia.chargeCoins(userId, coins);
+
+		// House has to make around 40% profit.
+		const profit = account.gambledWin * 0.4;
+
+		// Apply percentage error to get how close the house is to 40% profit.
+		const margin = Math.abs(( account.gambledLoss - account.gambledWin ) - profit) / profit;
+
+		// Calculate the chances of the user winning with the profit margin on top.
+		const stake = coins / ( multiplier * coins ) - margin;
```

Comment:
This can be simplified.
```suggestion
		const stake = 1 / multiplier - margin;
```

Reply:
`1`? That... Wouldn't work.

Reply:
It would, as long as `coins` is not `0`. Luckily you can't gamble `0` coins.

Reply:
I don't... Get how it would. Maybe you are not understanding the reason why it is done like this?

The way it's done is by diving the coins spent by the payout of the investment. With that you'll get the chances of that "coin flip" but we reduce the chance by a margin, called, profit margin which might need to be considerably amped up eventually.

Reply:
You probably should write out how this entire thing is mathematically supposed to work, since I have no idea and the code does not seem to reflect what you had in mind. (Or I am somehow misreading it)
Would make a great addition to the PR description.